### PR TITLE
add an unified "blob" cache

### PR DESCRIPTION
**Pull request by syjer**
with a configurable "directory", used by file upload, font and images for passkit.

### CODE REVIEW

File: src/main/java/alfio/manager/FileBlobCacheManager.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+@Component
+public class FileBlobCacheManager {
+
+    private static final Logger log = LoggerFactory.getLogger(FileBlobCacheManager.class);
+    private final String cacheDir;
+
+
+    public FileBlobCacheManager(Environment environment) {
+        this.cacheDir = Optional.ofNullable(environment.getProperty("alfio.cache-dir")).orElse(System.getProperty("java.io.tmpdir"));
```
Comment (cbellone) [inline src/main/java/alfio/manager/FileBlobCacheManager.java:42]: can be replaced with this overload: https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/core/env/PropertyResolver.html#getProperty(java.lang.String,java.lang.String)

File: src/main/java/alfio/manager/FileBlobCacheManager.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        Assert.isTrue(childPath.startsWith(parentPath), () -> "Resource path " + childPath + "must be inside the blob path " + parentPath);
+    }
+
+    public File getFile(String section, String id, Supplier<File> supplier) {
+        var resourcePath = getBlobDir(section).resolve(id);
+        checkPath(resourcePath, section);
+        if (Files.exists(resourcePath)) {
+            return resourcePath.toFile();
+        }
+        log.info("Cache not hit for file {}", id);
```
Comment (cbellone) [inline src/main/java/alfio/manager/FileBlobCacheManager.java:61]: better log cache miss at DEBUG or TRACE level?

**Review by cbellone — APPROVED** [review]

File: src/main/java/alfio/manager/FileBlobCacheManager.java (hunk 1 above)
Comment (syjer) [reply src/main/java/alfio/manager/FileBlobCacheManager.java]: fixed

File: src/main/java/alfio/manager/FileBlobCacheManager.java (hunk 2 above)
Comment (syjer) [reply src/main/java/alfio/manager/FileBlobCacheManager.java]: updated to debug

**Pull request merged**